src/workspaces/workspaces.service.ts

```diff
@@ -70,4 +86,153 @@ export class WorkspacesService {
     }
     return userWorkspace.workspace
   }
+
+  async update(userId:string, workspaceId:string, updateWorkspaceDto: UpdateWorkspaceDto){
+
+
+    const workspace = await this.workspaceRepository.findOne({
+        where: {
+          id: workspaceId
+        }
+    })
+
+    if(!workspace){
+      throw new NotFoundException("Workspace not found")
+    }
+
+    const userWorkspace = await this.userWorkspaceRepository.findOne({
+      where: {
+        userId,
+        workspaceId
+      }
+    })
+
+    if(!userWorkspace) {
+      throw new ForbiddenException('Dear User you are not allowed to edit this workspace')
+    }
+
+
+    await this.workspaceRepository.update({id: workspaceId},{name: updateWorkspaceDto.name})
+    return await this.workspaceRepository.findOne({
+      where: {
+        id:workspaceId
+    }})
+  }
+
+  async inviteUser(userId: string,workspaceId: string, payload:InviteUserDto){
+     const workspace = await this.workspaceRepository.findOne({
+        where: {
+          id: workspaceId
+        }
+    })
+
+    if(!workspace){
+      throw new NotFoundException("Workspace not found")
+    }
+
+    // Check if you are the owner of the workspace 
+
+      const userWorkspace = await this.userWorkspaceRepository.findOne({
+      where: {
+        userId,
+        workspaceId,
+        role: UserRole.ADMIN
+      }
+    })
+
+    if(!userWorkspace) {
+      throw new ForbiddenException('Dear User you are not allowed to Invite User to this workspace')
+    }
+
+
+    // Check if invited email is not admin
+
+    const adminUser = await this.userService.findOne(parseInt(userId))
+
+    if( adminUser && adminUser.email === payload.email){
+      throw new BadRequestException('You cannot invite yourself to a workspace')
+    }
+
+    const existingUser = await this.userService.findByEmail(payload.email)
+
+    if(existingUser) {
+      // Check if user exist in this workspace
+      const existingMember = await this.userWorkspaceRepository.findOne({
+        where: {
+          userId: String(existingUser.id),
+          workspaceId
+        }
+      })
+
+      if(existingMember) {
+        throw new ConflictException('This user already exist in this workspace')
+      }
+    }
+
```

Comment:
I would suggest to move this implementation into a function called something like `checkIfUserExists`